lib/device.c

```diff
@@ -240,6 +245,7 @@ struct sync_device *sync_create_device(const char *base)
 	d->io_cb.open = (void *(*)(const char *, const char *))fopen;
 	d->io_cb.read = (size_t (*)(void *, size_t, size_t, void *))fread;
 	d->io_cb.close = (int (*)(void *))fclose;
+	d->path_encode_cb = (const char *(*)(const char *path))path_encode;
```

Comment:
You shouldn't need the cast here. It's only there above to cast the FILE* part of the signature to void*.